### PR TITLE
sprint-7 release

### PR DESCRIPTION
**Pull request by lpendem**
This release PR includes the following stories completed in sprint-7:
* https://github.com/nciinc/fs-permit-platform/issues/149
* https://github.com/nciinc/fs-permit-platform/issues/162
* https://github.com/nciinc/fs-permit-platform/issues/198
* https://github.com/nciinc/fs-permit-platform/issues/207
* https://github.com/nciinc/fs-permit-platform/issues/211
* https://github.com/nciinc/fs-permit-platform/issues/212
* https://github.com/nciinc/fs-permit-platform/issues/218
* https://github.com/nciinc/fs-permit-platform/issues/236
* https://github.com/nciinc/fs-permit-platform/issues/238
* https://github.com/nciinc/fs-permit-platform/issues/239
* https://github.com/nciinc/fs-permit-platform/issues/240
* https://github.com/nciinc/fs-permit-platform/issues/241
* https://github.com/nciinc/fs-permit-platform/issues/257
* https://github.com/nciinc/fs-permit-platform/issues/258
* https://github.com/nciinc/fs-permit-platform/issues/259
* https://github.com/nciinc/fs-permit-platform/issues/276
* https://github.com/nciinc/fs-permit-platform/issues/283
* https://github.com/nciinc/fs-permit-platform/issues/284
* https://github.com/nciinc/fs-permit-platform/issues/285
* https://github.com/nciinc/fs-permit-platform/issues/286

### CODE REVIEW

File: frontend/src/app/_services/util.service.ts [hunk 1]
```diff
@@ -42,15 +41,13 @@ export class UtilService {
...
+  handleError(error: HttpErrorResponse | any) {
     let errors: any = [];
-    if (error instanceof Response) {
+    if (error instanceof HttpErrorResponse) {
       if (error.status) {
         switch (error.status) {
           case 400:
-            body = error.json() || '';
-            errors = body.errors;
+            errors = error.error.errors || '';
```
Comment (lauraGgit) [inline frontend/src/app/_services/util.service.ts]: is this how angular 5's `HttpErrorResponse` handles responses? That deeply nested?

File: frontend/src/app/application-forms/validators/error-message.component.ts [hunk 2]
```diff
@@ -11,6 +11,7 @@ export class ErrorMessageComponent {
   @Input() name: string;
 
   parseErrors(errors) {
+    console.log(errors);
```
Comment (lauraGgit) [inline frontend/src/app/application-forms/validators/error-message.component.ts]: Do you still need this debugging log?

File: frontend/src/app/trees/forests/forest-finder/remove-punc.pipe.ts [hunk 3]
```diff
@@ -4,13 +4,13 @@ import { Pipe, PipeTransform } from '@angular/core';
   name: 'RemovePunc'
 })
 export class RemovePuncPipe implements PipeTransform {
-
   transform(value: any, toRemove: Array<string>): any {
     let output = value;
-    toRemove.map((punc) => {
+    if (output) {
```
Comment (lauraGgit) [inline frontend/src/app/trees/forests/forest-finder/remove-punc.pipe.ts]: is this for breadcrumbs and routes?

File: frontend/e2e/authenticated/app.po.ts [hunk 4]
```diff
@@ -26,6 +26,10 @@ export class AdminApplicationList {
   navigateTo() {
     return browser.get('/admin/applications');
   }
+
+  permitId() {
+    return 'permit-806d3550-309d-46ea-b12a-f021f7b3d447';
```
[the remark this answers is not in the document]
Comment (lpendem) [reply frontend/e2e/authenticated/app.po.ts]: @lauraGgit this intake's application number from seeder used for tests.

File: frontend/src/app/application-forms/validators/error-message.component.ts (hunk 2 above)
Comment (lpendem) [reply frontend/src/app/application-forms/validators/error-message.component.ts]: @lauraGgit added task in sprint-8 story #102

File: frontend/src/app/trees/forests/forest-finder/remove-punc.pipe.ts (hunk 3 above)
Comment (lpendem) [reply frontend/src/app/trees/forests/forest-finder/remove-punc.pipe.ts]: @lauraGgit this is for routes, but likely dropped while implenting #102.

**Pull request merged**